### ISSUE TITLE
False positive on class definition keyword arguments

### ISSUE THREAD

**Issue by anuppari** [python/mypy#11036]
**Bug Report**

Mixing a numpy `ndarray` with a metaclass that uses keyword arguments causes a false positive.

**To Reproduce**
Run `mypy` on the following code

```python
from typing import Dict, Optional, Tuple
import numpy as np

class Meta(type):
    def __new__(
        cls, name: str, bases: Tuple[type, ...], dct: Dict, *args,
        a: Optional[int] = None, **kwargs
    ):
        return super().__new__(cls, name, bases, dct, *args, **kwargs)

class A(np.ndarray, a=1, metaclass=Meta):
    pass

a = A(shape=(2,))
```

**Expected Behavior**

Should pass without errors

**Actual Behavior**

Get the following error:
```
error: Unexpected keyword argument "a" for "__init_subclass__" of "object"
```

**Your Environment**

- Mypy version used: 0.910
- Mypy command-line flags: N/A
- Mypy configuration options from `mypy.ini` (and other config files): N/A
- Python version used: 3.8.6
- Operating system and version: Ubuntu 20.04

**Comment by rsnyman** [comment]
I am getting the same error, running MyPy with [SQLModel](https://sqlmodel.tiangolo.com/tutorial/fastapi/multiple-models/):

```python
class HeroBase(SQLModel):
    name: str = Field(index=True)
    secret_name: str
    age: Optional[int] = Field(default=None, index=True)


class Hero(HeroBase, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
```

And the error:

```
mypy: Unexpected keyword argument "table" for "__init_subclass__" of "object"  [call-arg] (E)
```

Python: 3.11.2
mypy: 1.0.0 (compiled: yes)
Operating system: Debian unstable

**Comment by AlexanderSalge** [comment]
Any way to disable this rule until it's fixed?

**Comment by erictraut** [comment]
I no longer see these errors with the recently-released mypy 1.5, so I suspect it has been fixed.

**Comment by YPCrumble** [comment]
@ilevkivskyi @AlexanderSalge I am still seeing this using MyPy 1.5.1 - are we sure this isn't still an issue?

**Comment by AlexanderSalge** [comment]
yeah, just updated everything. I'm still seeing it.

**Comment by ilevkivskyi** [comment]
I cannot reproduce it on Python 3.10

**Comment by AlexanderSalge** [comment]
running in 3.11 and getting the issue with sqlmodel as @rsnyman mentioned

**Comment by german-muzquiz** [comment]
Still getting this issue with mypy `1.7.0`

**Comment by jumanjiman** [comment]
i'm now seeing this with:

- sqlmodel 0.0.14
- mypy 1.7.1
- mypy-extensions  1.0.0
- SQLAlchemy 2.0.23
- pydantic 1.10.13

```
src/contrib/postgres/postgres_iam_schema.py:38: error: Unexpected keyword argument "table" for "__init_subclass__" of "object"  [call-arg]
/root/.cache/pre-commit/repoi8e7rml7/py_env-python3/lib/python3.11/site-packages/mypy/typeshed/stdlib/builtins.pyi:124: note: "__init_subclass__" of "object" defined here
src/contrib/postgres/postgres_iam_schema.py:38: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
src/contrib/postgres/postgres_iam_schema.py:165: error: Unexpected keyword argument "table" for "__init_subclass__" of "object"  [call-arg]
/root/.cache/pre-commit/repoi8e7rml7/py_env-python3/lib/python3.11/site-packages/mypy/typeshed/stdlib/builtins.pyi:124: note: "__init_subclass__" of "object" defined here
src/contrib/postgres/postgres_iam_schema.py:165: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
src/contrib/postgres/postgres_cloud_approvals_schema.py:35: error: Unexpected keyword argument "table" for "__init_subclass__" of "object"  [call-arg]
/root/.cache/pre-commit/repoi8e7rml7/py_env-python3/lib/python3.11/site-packages/mypy/typeshed/stdlib/builtins.pyi:124: note: "__init_subclass__" of "object" defined here
src/contrib/postgres/postgres_cloud_approvals_schema.py:35: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
Found 6 errors in 2 files (checked 159 source files)
```

**Comment by EwertonDCSilv** [comment]
@tiangolo alguma ideia de como contornar esse problema ?

**Comment by EwertonDCSilv** [comment]
https://github.com/python/mypy/issues/11036#issuecomment-1845812684
@jumanjiman tenho o mesmo problema que você com Python 3.12.0

```
aiofiles==23.2.1

pydantic==1.10.13

fastapi==0.105.0
httpx==0.25.2
responses==0.24.1
coverage==7.3.3
pytest==7.4.3
pytest-cov==4.1.0
pytest-httpx==0.27.0
pytest-asyncio==0.23.2
mock==5.1.0
asyncpg==0.29.0
sqlmodel==0.0.14
SQLAlchemy==2.0.23  
alembic==1.13.0
psycopg2-binary==2.9.9
python-magic==0.4.27
structlog==23.2.0
celery==5.3.6
celery[redis]==5.3.6
celery[librabbitmq]==5.3.6
```

**Comment by EwertonDCSilv** [comment]
Existe um topico do SqlModel para o assunto 
https://github.com/tiangolo/sqlmodel/discussions/732

**Comment by ndellosa95** [comment]
Still seeing this in mypy 1.9.0 with Python 3.11.4.

**Comment by YoniMelki** [comment]
Still seeing this with:

* mypy = 1.13.0
* pydantic = 2.9.2
* sqlmodel = 0.0.22
* python version = 3.12.4

**Comment by Cuick** [comment]
Adding an __init__.py file to the relevant directory can resolve this issue.
